# 2. Probabilidade condicional e teorema de Bayes

## 2.1. Probabilidade condicional

De forma geral, probabilidade condicional repsenta a probabilidade de se observar um resultado sob alguma condição. Vamos defini-la e depois avaliar alguns exemplos. Suponha que desejamos calcular a probabilidade de se observar o evento $A$ sendo que já sabemos que $B$ é ocorreu, então:

\begin{equation}
    \mathbb{P}(A|B) = \frac{\mathbb{P}(A \cap B)}{\mathbb{P}(B)}
\end{equation}

Em que $\mathbb{P}(A|B)$ é lido como: probabilidade de $A$ dado $B$. Enquanto $\mathbb{P}(A \cap B)$ é a probabilidade de se observar $A$ e $B$, ou seja, a intersecção entre os dois eventos. Note que a equação acima só é válida se $\mathbb{P}(B) > 0$.

### Exemplo

Retomando o exemplo do lançamento de um dado, temos o espaço amostral $\Omega = \{1, 2,3,4,5,6\}$. Suponha que sabemos que o dado foi lançado e se observou um número maior que $3$, então qual a probabilidade do número observado ter sido par. Temos duas abordagens para resolver o problema:
1. A mais direta e sem fórmula parte da observação de que, como sabemos que o resultado foi maior que $3$ então temos como possíveis resultados $\{4, 5, 6\}$. Logo, os números pares nesse conjunto são $\{4, 6\}$. Daí, temos que dois resultados em três são pares, ou: $\mathbb{P}(P|M) = 2/3$.
2. A segunda abordagem parte da fórmula acima, note que $\mathbb{P}(P \cap M) = \mathbb{P}(\{2,4,6\} \cap \{ 4, 5, 6 \}) = \mathbb{P}(\{4, 6\})$ = 2/6. Além disso, $\mathbb{P}(M) = \mathbb{P}(\{4, 5, 6\}) = 3/6 = 1/2$. Aplicando a fórmula anterior: $\mathbb{P}(P|M) = \mathbb{P}(P \cap M)/\mathbb{P}(M) = (2/6)/(1/2) = 2/3$.

## 2.2. Lei da probabilidade total

Considere que o espaço amostral possa ser dividido em subconjuntos (ou eventos): $B_1, B_2, B_3, ..., B_n$. E estejamos interessados na probabilidade de um evento $A$, então a seguinte equação é válida:

\begin{equation}
    \mathbb{P}(A) = \mathbb{P}(A|B_1)\mathbb{P}(B_1) + \mathbb{P}(A|B_2)\mathbb{P}(B_2) + ... + \mathbb{P}(A|B_n)\mathbb{P}(B_n)
\end{equation}

Podemos entender essa fórmula da seguinte maneira: a probabilidade de se observar o evento $A$ é igual á probabilidade de se observar $A$ condicional a $B_1$ ter ocorrido, multiplicado pela probabilidade de $B_1$ mais a probabilidade de $A$ condicional à $B_2$ etc.

### Exemplo

Voltando ao exemplo do dado. Note que os possíveis resultados podem ser divididos em pares e ímpares. Agora, qual a probabilidade de se observar um resultado maior que 3? Já vimos como responder à isso de outras maneiras, mas por fins didáticos, se usarmos a lei da probabilidade total encontramos: $\mathbb{P}(M) = \mathbb{P}(M|P)\mathbb{P}(P) + \mathbb{P}(M|I)\mathbb{P}(I)$.

Note que:
- $\mathbb{P}(M|P) = \mathbb{P}(M \cap P)/\mathbb{P}(P) = \mathbb{P}(\{4, 6\})/\mathbb{P}(\{2, 4, 6\}) = 2/3$
- $\mathbb{P}(M|I) = \mathbb{P}(M \cap I)/\mathbb{P}(I) = \mathbb{P}(\{5\})/\mathbb{P}(\{1, 3, 5\}) = 1/3$
- $\mathbb{P}(P) = 1/2$
- $\mathbb{P}(I) = 1/2$

Aplicando a fórmula anterior encontramos $\mathbb{P}(M) = (2/3)(1/2) + (1/3)(1/2) = 1/2$.

## 2.3. Teorema de Bayes

A probabilidade condicional é peça fundamental de um dos resultados mais importantes da estatística: o teorema de Bayes. Em geral durante pesquisas científicas ou investigações nós formulamos hipóteses. Conforme coletamos mais dados, ou temos uma nova descoberta, precisamos atualizar nossa probabilidade. Esse é o problema que entenderemos como resulver. Partiremos da fórmula:

\begin{equation}
    \mathbb{P}(A | B) = \frac{\mathbb{P}(B | A) \mathbb{P}(A)}{\mathbb{P}(B)}
\end{equation}

Nesse caso, a partir de uma probabilidade condicional $\mathbb{P}(B|A)$, temos um caminho para encontrar $\mathbb{P}(A|B)$. Para ficar mais claro, considere que $A$ é uma hpótese de um pesquisador. Com base em suas pesquisas, a probabilidade dessa hipótese estar correta é $\mathbb{P}(A)$. Com o tempo, e realização de experimentos ele se depara com um resultado $B$, que ocorre com probabilidade $\mathbb{P}(B)$. Além disso, ele sabe que se a hipótese dele ($A$) estiver correta, então a probabilidade de se observar $B$ é $\mathbb{P}(B|A)$.

### Exemplo

Um dos exemplos mais clássicos sobre a aplicação do teorema de Bayes refere-se ao resultado de exames médicos. Infelizmente os exames estão sujeitos a erros:
- Falso positivo: resultado do exame é positivo, apesar da pessoa não ter a doença,
- Falso negativo: resultado do exame é negativo, apesar da pessoa ter a doença.

Considere o seguinte problema: uma doença ocorre na população com probabilidade 5%. Para as pessoas que tem a doença, a probabilidade de se observar um resultado positivo é 90%, e negativo é 10% (*falso negativo*). Por outro lado, para quem não tem a doença, a probabilidade de ter um resultado positivo é 4% (falso positivo), enquanto a probabilidade de ter resltado negativo é 96%. Podemos resumir isso na tabela:

|Teste | Doente (5%) | Não doente (95%) |
|------|-------------|------------------|
|Positivo| 90% | 4%|
|Negativo| 10% |96%|

Nesse caso, qual a probabilidade da pessoa estar doente, sendo que o resultado do exame foi positivo?

Note que queremos usar a fórmula de Bayes. Adaptando $A$ e $B$ para nossos eventos, temos:

\begin{equation*}
    \mathbb{P}(D|+) = \frac{\mathbb{P}(+|D) \mathbb{P}(D)}{\mathbb{P}(+)}
\end{equation*}

Em que $D$ representa pessoa doente, e $+$ representam resultado positivo. Note que já temos alguns dos dados pelo enunciado:
- A probabilidade de uma pessoa ter um resultado positivo sendo que ela está doente é: $\mathbb{P}(+|D) = 90\% = 0.9$
- A probabilidade de uma pessoa ter a doença é $\mathbb{P}(D) = 5\% = 0.05$

Por fim, precisamos de uma última informação, a probabilidade de uma pessoa ter resultado positivo. Nesse ponto, usamos a lei da probabilidade total. Note que existem duas possibilidades que resultam em teste positivo: "pessoa doente e resultado positivo" ou "pessoa não doente e resultado positivo". Teríamos:

\begin{equation*}
    \mathbb{P}(+) = \mathbb{P}(+|D)\mathbb{P}(D) + \mathbb{P}(+|ND)\mathbb{P}(ND)
\end{equation*}

Em que $ND$ é não doente, atualizando os valores, temos:

\begin{equation*}
    \mathbb{P}(+) = 0.9 \times 0.05 + 0.04 \times 0.95
\end{equation*}

Usando ``python`` para fazer os cálculos:

In [1]:
p_positivo_doente = 0.9
p_doente = 0.05
p_positivo = (0.9*0.05 + 0.04*0.95)

p_doente_positivo = p_positivo_doente*p_doente/p_positivo

print(f"A probabilidade de uma pessoa estar doente, sendo que ela testou positivo, é: {p_doente_positivo:.4f}")

A probabilidade de uma pessoa estar doente, sendo que ela testou positivo, é: 0.5422


Logo, a probabilidade da pessoa estar doente, sendo que seu teste deu positivo é de 54%.